core/app/models/spree/taxon.rb

```diff
@@ -95,6 +98,23 @@ def child_index=(idx)
       move_to_child_with_index(parent, idx.to_i) unless new_record?
     end
 
+    def move_to_child_with_index(node, index)
+      # closure_tree does not provide any method that move nodes regarding position
+      # this method overrides move_to_child_with_index offered by awesome_nested_set
```

Comment:
@mamhoff I fixed the corresponding tests. [Here](https://github.com/solidusio/solidus/blob/master/api/app/controllers/spree/api/taxons_controller.rb#L10) I ordered the taxons by `permalink`, avoiding a complex inner join into the hierarchy table, but if you have any other approach please let me know!